docs/en/02_Developer_Guides/18_Cookies_And_Sessions/01_Cookies.md

```diff
@@ -52,6 +56,44 @@ Cookie::force_expiry($name, $path = null, $domain = null);
 // Cookie::force_expiry('MyApplicationPreference')
 ```
 
+### Samesite attribute
+
+The `samesite` attribute is set on all cookies with a default value of `Lax`. You can change the default value by setting the `default_samesite` value on the
+[Cookie](api:SilverStripe\Control\Cookie) class:
+
+```yml
+SilverStripe\Control\Cookie:
+  default_samesite: 'Strict'
+```
+
+If you need to set the `samesite` attribute for a specific cookie, you can implement the `updateSameSite()` method on an `Extension` subclass and apply that to
+[CookieJar](api:SilverStripe\Control\CookieJar) - though note that this extension method will stop working in 5.0 in favour of a new parameter for passing the
+`samesite` attribute value directly.
+
+```yml
+SilverStripe\Control\CookieJar:
+  extensions:
+    - App\Extension\CookieJarExtension
+```
+
+```php
+<?php
+
+use SilverStripe\Core\Extension;
```

Comment:
Why are we using an extension here instead of just adding an extra param? i.e. why would we need to wait for a new CMS version?  Seems bad to say "this is how you do the thing" and immediately deprecate it.

Reply:
Adding an extra parameter will break any custom classes that implement `Cookie_Backend`.

Reply:
OK, makes sense that we're constrained by the interface

I'm inclined to just get rid of the extension hook functionality altogether and just not allow overriding on individual cookie names. I don't like the idea of adding functionality only to deprecate it straight away.  It just makes upgrading more painful.

Currently people are already using 'Lax' due to it being a browser default, so by not providing this functionality we are not going to break any implementations.

Reply:
That's fair. In a major release the deprecation notice there should prompt us to add the parameters, so people can set different `samesite` attributes to their specific cookies that way. In the meantime being able to change the value globally is still more control than people currently have so I'm okay with not providing the extension.
If someone does really want this then hopefully they'll create an issue or a PR and further discussion can be had at that time.